### ISSUE TITLE
Data delivery issues 

### ISSUE THREAD

**Issue by qoqobolo** [status-im/status-react#16873]
# Bug Report

There are three most obvious data delivery issues that don't have exact steps to reproduce but that we encounter quite often. 
As discussed, describing them here for further investigation (cc @siphiuel )
Logs and other details will be added as any of these are encountered.

**1. Data loss on a synced device**
Most often this can be observed in group chats (as described in https://github.com/status-im/status-mobile/issues/16786 with more details) or in Status Community.

**2. After background on iOS**
Sometimes we face the situation when data (messages/contact or membership requests/etc) stops being sent/received on iOS device after some time in background.
To start receiving/sending data again, you need to reopen the app and/or relogin.

**3. Significant delivery delays in the late afternoon (for European users)**
Sometimes the delivery of messages/requests becomes especially long in the afternoon compared to the morning for time zones such as UTC+2, UTC +3.

**4. Some particular kind of data stops being delivered for no apparent reason**
For example, you still can receive messages and contact requests, but stop getting approved to join the community.
Or you can receive messages but can't receive contact requests.
Sometimes relogin works, sometimes it doesn't. 

#### _Note_
Some of these scenarios may actually be the same thing. Unfortunately, it is not always possible to distinguish between a delay in delivery and a complete loss of data.

**Comment by qoqobolo** [comment]
Example with a contact request from e2e tests:
1. UserA sends CR to UserB
2. UserB accepts CR

**Actual result:** 
UserB does not appear in UserA's contacts

[cr_sender_admin.log](https://github.com/status-im/status-mobile/files/12252665/cr_sender_admin.log)
[cr_receiver_member_2.log](https://github.com/status-im/status-mobile/files/12252667/cr_receiver_member_2.log)

**Comment by pavloburykh** [comment]
> 

Another example of CR not delivered. Currently it is causing the whole e2e group of tests to fail. 

[CR_receiver_logcut.log](https://github.com/status-im/status-mobile/files/12280689/CR_receiver_logcut.log)
[CR_sender_logcat.log](https://github.com/status-im/status-mobile/files/12280690/CR_sender_logcat.log)

Here is a video from saucelab

Device 1 sends CR to Device 2 https://app.eu-central-1.saucelabs.com/tests/70c14d8c67c94b509409174a9d107f85?auth=5b7b8a386f5c29002365977c4ac0bb1f#52

Device 2 does not receive CR https://app.eu-central-1.saucelabs.com/tests/b71969cb18bd4787866cac6a1068d9b5?auth=ec1bb0869ff82b5b6049c79f280834b0#55

waiting time: 60 sec

**Comment by cammellos** [comment]
@pavloburykh do you have the geth.log for both sender/receiver please?

**Comment by pavloburykh** [comment]
> @pavloburykh do you have the geth.log for both sender/receiver please?

hi @cammellos! Unfortunately all we have are the above attached logcat files.

**Comment by churik** [comment]
One more issue with data delivery which was found and investigated during PR testing - more context is here: https://github.com/status-im/status-mobile/pull/17091#issuecomment-1693280668

**Comment by yevh-berdnyk** [comment]
Example of missing a message in a community channel after user unblock, which causes e2e test fail in  about 10% runs:

1. User A blocks user B
2. User A unblocks user B
3. User B sends a message 'Hurray! unblocked' in a community channel

SauceLabs session of user A: https://app.eu-central-1.saucelabs.com/tests/b263360bc45f418c9a108c656b5bb208?auth=1443824e95e223f47a2eac9dc6161cfc#553
SauceLabs session of user B: https://app.eu-central-1.saucelabs.com/tests/c1d02548696049bb9aed4b691d7c3e02?auth=a6476d1fecd4aa06e67331d099632cea#375

Logs:
[user_A_geth.log](https://github.com/status-im/status-mobile/files/12586420/user_A_geth.log)
[user_B_geth.log](https://github.com/status-im/status-mobile/files/12586421/user_B_geth.log)

[user_A_logcat.log](https://github.com/status-im/status-mobile/files/12586422/user_A_logcat.log)
[user_B_logcat.log](https://github.com/status-im/status-mobile/files/12586423/user_B_logcat.log)

**Comment by vitvly** [comment]
> Example of missing a message in a community channel after user unblock, which causes e2e test fail in about 10% runs:
> 
> 1. User A blocks user B
> 2. User A unblocks user B
> 3. User B sends a message 'Hurray! unblocked' in a community channel
> 
> SauceLabs session of user A: https://app.eu-central-1.saucelabs.com/tests/b263360bc45f418c9a108c656b5bb208?auth=1443824e95e223f47a2eac9dc6161cfc#553 SauceLabs session of user B: https://app.eu-central-1.saucelabs.com/tests/c1d02548696049bb9aed4b691d7c3e02?auth=a6476d1fecd4aa06e67331d099632cea#375
> 
> Logs: [user_A_geth.log](https://github.com/status-im/status-mobile/files/12586420/user_A_geth.log) [user_B_geth.log](https://github.com/status-im/status-mobile/files/12586421/user_B_geth.log)
> 
> [user_A_logcat.log](https://github.com/status-im/status-mobile/files/12586422/user_A_logcat.log) [user_B_logcat.log](https://github.com/status-im/status-mobile/files/12586423/user_B_logcat.log)

In this scenario User A does not receive a message, and in their geth.log there's lots of `could not add wakuv2 filter for peer` messages:
```
53591:t=2023-09-12T02:38:52+0000 lvl=warn msg="could not add wakuv2 filter for peer" peer=16Uiu2HAm1fVVprL9EaDpDw4frNX3CPfZu5cBqhtk9Ncm6WCvprpv
53600:t=2023-09-12T02:38:52+0000 lvl=warn msg="could not add wakuv2 filter for peer" peer=16Uiu2HAm1fVVprL9EaDpDw4frNX3CPfZu5cBqhtk9Ncm6WCvprpv
53610:t=2023-09-12T02:38:52+0000 lvl=warn msg="could not add wakuv2 filter for peer" peer=16Uiu2HAm1fVVprL9EaDpDw4frNX3CPfZu5cBqhtk9Ncm6WCvprpv
53913:t=2023-09-12T02:38:56+0000 lvl=warn msg="could not add wakuv2 filter for peer" peer=16Uiu2HAm1fVVprL9EaDpDw4frNX3CPfZu5cBqhtk9Ncm6WCvprpv
53917:t=2023-09-12T02:38:56+0000 lvl=warn msg="could not add wakuv2 filter for peer" peer=16Uiu2HAm1fVVprL9EaDpDw4frNX3CPfZu5cBqhtk9Ncm6WCvprpv
54047:t=2023-09-12T02:39:01+0000 lvl=warn msg="could not add wakuv2 filter for peer" peer=16Uiu2HAm1fVVprL9EaDpDw4frNX3CPfZu5cBqhtk9Ncm6WCvprpv
```
This happens when peer count dropped to zero and full resubscribe is in order. Also at the same time lots of `could not unsubscribe from peer`:
```
53419:t=2023-09-12T02:38:52+0000 lvl=eror msg="could not unsubscribe from peer" peerID=16Uiu2HAkyScd7DiwgMwzfw8CFFhznH3wRzciqEUfjDzn7vyimR8c error="failed to dial: failed to dial 16Uiu2HAkyScd7DiwgMwzfw8CFFhznH3wRzciqEUfjDzn7vyimR8c:\n  * [/ip4/134.209.135.195/tcp/30303] dial backoff"
53435:t=2023-09-12T02:38:52+0000 lvl=eror msg="could not unsubscribe from peer" peerID=16Uiu2HAkyScd7DiwgMwzfw8CFFhznH3wRzciqEUfjDzn7vyimR8c error="failed to dial: failed to dial 16Uiu2HAkyScd7DiwgMwzfw8CFFhznH3wRzciqEUfjDzn7vyimR8c:\n  * [/ip4/134.209.135.195/tcp/30303] dial backoff"
```

Message from UserB sent on `02:41:38`:
```
user_B_geth.log
t=2023-09-12T02:41:38+0000 lvl=dbug msg=calculateSuggestions chatID=0x02bdd8711ce4de6f64cde9349e86df11b49f1ce9e817149e403957f1ac867b4d5ce8ab646d-2d6d-4a61-94ab-71fa65e71b51 fullText="Hurray! unblocked" num of mentionable user=1
```
 Also there are lots of re-subscriptions to the same peer:
```
65312:t=2023-09-12T02:43:22+0000 lvl=info msg="wakuv2 filter subscription success" peer=16Uiu2HAm1fVVprL9EaDpDw4frNX3CPfZu5cBqhtk9Ncm6WCvprpv pubsubTopic=/waku/2/default-waku/proto contentTopics=[/waku/1/0x88ff4997/rfc26]
65688:t=2023-09-12T02:43:25+0000 lvl=info msg="wakuv2 filter subscription success" peer=16Uiu2HAm1fVVprL9EaDpDw4frNX3CPfZu5cBqhtk9Ncm6WCvprpv pubsubTopic=/waku/2/default-waku/proto contentTopics=[/waku/1/0x6ba82a67/rfc26]
65708:t=2023-09-12T02:43:25+0000 lvl=info msg="wakuv2 filter subscription success" peer=16Uiu2HAm1fVVprL9EaDpDw4frNX3CPfZu5cBqhtk9Ncm6WCvprpv pubsubTopic=/waku/2/default-waku/proto contentTopics=[/waku/1/0xde6e06b5/rfc26]
65844:t=2023-09-12T02:43:32+0000 lvl=info msg="wakuv2 filter subscription success" peer=16Uiu2HAm1fVVprL9EaDpDw4frNX3CPfZu5cBqhtk9Ncm6WCvprpv pubsubTopic=/waku/2/default-waku/proto contentTopics=[/waku/1/0x6ba82a67/rfc26]
66043:t=2023-09-12T02:43:42+0000 lvl=info msg="wakuv2 filter subscription success" peer=16Uiu2HAm1fVVprL9EaDpDw4frNX3CPfZu5cBqhtk9Ncm6WCvprpv pubsubTopic=/waku/2/default-waku/proto contentTopics=[/waku/1/0x6ba82a67/rfc26]
66356:t=2023-09-12T02:43:42+0000 lvl=info msg="wakuv2 filter subscription success" peer=16Uiu2HAm1fVVprL9EaDpDw4frNX3CPfZu5cBqhtk9Ncm6WCvprpv pubsubTopic=/waku/2/default-waku/proto contentTopics=[/waku/1/0x6ba82a67/rfc26]
66357:t=2023-09-12T02:43:42+0000 lvl=info msg="wakuv2 filter subscription success" peer=16Uiu2HAm1fVVprL9EaDpDw4frNX3CPfZu5cBqhtk9Ncm6WCvprpv pubsubTopic=/waku/2/default-waku/proto contentTopics=[/waku/1/0xde6e06b5/rfc26]
66570:t=2023-09-12T02:43:52+0000 lvl=info msg="wakuv2 filter subscription success" peer=16Uiu2HAm1fVVprL9EaDpDw4frNX3CPfZu5cBqhtk9Ncm6WCvprpv pubsubTopic=/waku/2/default-waku/proto contentTopics=[/waku/1/0xde6e06b5/rfc26]
67008:t=2023-09-12T02:43:58+0000 lvl=info msg="wakuv2 filter subscription success" peer=16Uiu2HAm1fVVprL9EaDpDw4frNX3CPfZu5cBqhtk9Ncm6WCvprpv pubsubTopic=/waku/2/default-waku/proto contentTopics=[/waku/1/0x6ba82a67/rfc26]
67025:t=2023-09-12T02:43:59+0000 lvl=info msg="wakuv2 filter subscription success" peer=16Uiu2HAm1fVVprL9EaDpDw4frNX3CPfZu5cBqhtk9Ncm6WCvprpv pubsubTopic=/waku/2/default-waku/proto contentTopics=[/waku/1/0xde6e06b5/rfc26]
```

**Comment by churik** [comment]
Starting from 22/09/23 we have significant delays in messages, which affects a lot of test runs:

https://github.com/status-im/status-mobile/pull/17346#issuecomment-1730797926
https://github.com/status-im/status-mobile/pull/17366#issuecomment-1731143233
https://github.com/status-im/status-mobile/pull/17385#issuecomment-1731179628


Example of failed group
Testrail: https://ethstatus.testrail.net/index.php?/tests/view/3238732


### Sender
Video: https://app.eu-central-1.saucelabs.com/tests/41965a93a2324dd2a33a0bb1aa56faa8?auth=4cba9affd39fd7311709b012df8c0723#60 - at 06:03 accepted PR
Logcat: [logcat (28).log](https://github.com/status-im/status-mobile/files/12700318/logcat.28.log)
Geth: [TestCommunityMultipleDeviceMerged_geth1.log](https://github.com/status-im/status-mobile/files/12700323/TestCommunityMultipleDeviceMerged_geth1.log)


### Receiver
Video: https://app.eu-central-1.saucelabs.com/tests/763d7a3a3b4a4ce4ab8eb3a482bc80a1?auth=b7d98bb23d80691bf637665d470ee49c#75 - wait for 90m sec to 07:30 - no new chat
Logcat: [logcat (27).log](https://github.com/status-im/status-mobile/files/12700312/logcat.27.log)
Geth:[TestCommunityMultipleDeviceMerged_geth0.log](https://github.com/status-im/status-mobile/files/12700315/TestCommunityMultipleDeviceMerged_geth0.log)

cc @siphiuel

**Comment by vitvly** [comment]
So in context of checking https://github.com/status-im/status-mobile/pull/17344 so far was able to find out that filter subscription liveness checks fail for some reason, which did not happen during testing in go-waku and status-go. Investigating.

EDIT: Also libp2p's SupportsProtocols fail on fleet nodes.

**Comment by siphiuel** [comment]
Also libp2p's `SupportsProtocols` fail on fleet nodes.

**Comment by vitvly** [comment]
Actually liveness checks do fail during testing as well, here's an example for PeerID=`16Uiu2HAmHKj9KTUEUPpw9F3EaDkT6QVXZNTRVerFJJtnkcC5CHgx`, logged when running `TestWakuV2Filter`:
```
2023-09-26T17:54:12.044+0300    INFO    wakuv2/waku.go:1017     wakuv2 filter subscription success      {"peer": "16Uiu2HAmHKj9KTUEUPpw9F3EaDkT6QVXZNTRVerFJJtnkcC5CHgx", "pubsubTopic": "/waku/2/default-waku/proto", "contentTopics": ["/waku/1/0x01020304/rfc26"]}

2023-09-26T17:54:12.930+0300    INFO    wakuv2/waku.go:1077     Check sub aliveness     {"peerId": "16Uiu2HAmHKj9KTUEUPpw9F3EaDkT6QVXZNTRVerFJJtnkcC5CHgx", "topics": ["/waku/1/0x01020304/rfc26"], "subID": "3340285d-5be1-4967-8600-b0d1ab6ffbf4"}

2023-09-26T17:54:12.987+0300    ERROR   node2.filterv2-lightnode        filter/client.go:213    receiving FilterSubscribeResponse {"error": "EOF"}
github.com/waku-org/go-waku/waku/v2/protocol/filter.(*WakuFilterLightNode).request
        /Users/vvlasov/c/status-go/vendor/github.com/waku-org/go-waku/waku/v2/protocol/filter/client.go:213
github.com/waku-org/go-waku/waku/v2/protocol/filter.(*WakuFilterLightNode).Ping
        /Users/vvlasov/c/status-go/vendor/github.com/waku-org/go-waku/waku/v2/protocol/filter/client.go:360
github.com/waku-org/go-waku/waku/v2/protocol/filter.(*WakuFilterLightNode).IsSubscriptionAlive
...
2023-09-26T17:54:12.987+0300    INFO    wakuv2/waku.go:1080     Sub not alive   {"peerId": "16Uiu2HAmHKj9KTUEUPpw9F3EaDkT6QVXZNTRVerFJJtnkcC5CHgx", "subID": "3340285d-5be1-4967-8600-b0d1ab6ffbf4", "error": "EOF"}
```

**Comment by vitvly** [comment]
Some updates:
  - filter liveness ping requests were missing a `requestId`, and nim-waku nodes threw "Failed to decode filter subscribe request" error, hence EOFs on our end. @richard-ramos identified and fixed it in https://github.com/waku-org/go-waku/pull/776
  - this did not happen on go-waku nodes, that is why it was actually possible from time to time to receive messages
  - after investigation in https://github.com/status-im/infra-status/issues/31 filter protocol was enabled on status-prod fleet in 
https://github.com/status-im/infra-status/commit/1fb6de29

Updated https://github.com/status-im/status-mobile/pull/17344 and related status-go with latest go-waku to verify on mobile.

**Comment by pavloburykh** [comment]
Our E2E sometimes catching fails of contact request delivery. 

Here is an example of e2e video were receiver has not received CR after 30 seconds of waiting
https://app.eu-central-1.saucelabs.com/tests/3698eaee3a044cfd8ffb51e642130d2c?auth=0164b52f82c9a68eab5f3cfd1a8d308a#56

Currently there are no steps for manual reproduction, but  maybe logs will help to identiify the problem 

[CR_ receiver.log](https://github.com/status-im/status-mobile/files/12847478/CR_.receiver.log)
[CR_sender.log](https://github.com/status-im/status-mobile/files/12847479/CR_sender.log)

cc @siphiuel

**Comment by vitvly** [comment]
@pavloburykh looked through logs. Haven't yet managed to precisely locate the issue. There were several stream resets in logs, but these should have been handled properly.

Identified one related problem here: https://github.com/status-im/status-go/pull/4131. Peer ping was not properly working for newly installed filters. Testing it now.

There's another weird problem: aliveness check failures with error message "404 - NOT_FOUND: peer has no subscriptions". Checked one particular case, and the peer in question was not unsubscribed from before the error, so i don't understand what is the problem yet. **UPDATE**: saw in the logs that the peer was disconnected earlier.

**Comment by pavloburykh** [comment]
> @pavloburykh looked through logs.

@siphiuel thank you for looking at the issue. Feel free to ping me if any help is needed. 

I am not sure it will help, but I will leave here some fresh logs from today's failed e2e. 

[CR_receiver.log](https://github.com/status-im/status-mobile/files/12857453/CR_receiver.log)
[CR_sender.log](https://github.com/status-im/status-mobile/files/12857454/CR_sender.log)

**Comment by yevh-berdnyk** [comment]
One more issue from today's nightly e2e.

User A sends messages in a community channel: https://app.eu-central-1.saucelabs.com/tests/78ccc8ee06b14f14bb79adad07c2de8c?auth=456033258b6e0da2dfb217dd5daf0dec#189
User B checks these messages: https://app.eu-central-1.saucelabs.com/tests/7d89f3a1d6d14f1f956664b26d426479?auth=bdf0b151e054af945c13d6c621cd8230#232

Sender logs: 
[test_community_markdown_support_geth2.log](https://github.com/status-im/status-mobile/files/13236503/test_community_markdown_support_geth2.log)
[logcat_sender.log](https://github.com/status-im/status-mobile/files/13236504/logcat_sender.log)



Receiver logs: 
[test_community_markdown_support_geth1.log](https://github.com/status-im/status-mobile/files/13236483/test_community_markdown_support_geth1.log)
[logcat_receiver.log](https://github.com/status-im/status-mobile/files/13236508/logcat_receiver.log)

**Comment by vitvly** [comment]
> One more issue from today's nightly e2e.
> 
> User A sends messages in a community channel: https://app.eu-central-1.saucelabs.com/tests/78ccc8ee06b14f14bb79adad07c2de8c?auth=456033258b6e0da2dfb217dd5daf0dec#189 User B checks these messages: https://app.eu-central-1.saucelabs.com/tests/7d89f3a1d6d14f1f956664b26d426479?auth=bdf0b151e054af945c13d6c621cd8230#232
> 
> Sender logs: [test_community_markdown_support_geth2.log](https://github.com/status-im/status-mobile/files/13236503/test_community_markdown_support_geth2.log) [logcat_sender.log](https://github.com/status-im/status-mobile/files/13236504/logcat_sender.log)
> 
> Receiver logs: [test_community_markdown_support_geth1.log](https://github.com/status-im/status-mobile/files/13236483/test_community_markdown_support_geth1.log) [logcat_receiver.log](https://github.com/status-im/status-mobile/files/13236508/logcat_receiver.log)

So far checked lightpush/filter operation, haven't noticed anything suspicious. Will check fleet logs.

**Comment by vitvly** [comment]
Sorry for late update. 2 things:
  - nim-waku fleet was updated to include some extra flags for a single node in order to facilitate message debugging: https://github.com/status-im/infra-status/issues/35
  - go-waku nodes will have some more debugging too: https://github.com/waku-org/go-waku/pull/886. Not deployed too.

This is the outcome of investigation carried out here https://github.com/status-im/status-mobile/issues/16873#issuecomment-1794401408 - seems that messages were somehow lost in transition. Therefore we need more clarity on what happens to particular message hashes once inside the fleet.

**Comment by pavloburykh** [comment]
@vitvly recently some e2e started failing because CR remains in Pending state for sender despite it has been Accepted on receiver's side. 

Here is an example from nightly run (17/11/23):

User A sends CR to User B https://app.eu-central-1.saucelabs.com/tests/5212df00ea3c4a0eb9f538d5670eff9b?auth=870cc82c3c2ccb46d7f905e2eff83d0b#101

User B receives and accepts CR https://app.eu-central-1.saucelabs.com/tests/8bab20c7b65f4712a861e750b11cc4fe?auth=2752805c9a47b849862f77848af559c1#63 

**Actual result:** CR remains in pending for sender User A. 

[CR_receiver_User_B.log](https://github.com/status-im/status-mobile/files/13391548/CR_receiver_User_B.log)
[CR_sender_User_A.log](https://github.com/status-im/status-mobile/files/13391549/CR_sender_User_A.log)

**Expected result:** CR state  changes to Accepted.

**Comment by pavloburykh** [comment]
Hello @vitvly!

Here is an example of messages are not delivered in community after receiver has come back online. 

On the video below you can observe the following: 

**Preconditions:** User A (Android) and User B (IOS) are members of the same community

1. User B and User A open the same community channel
2. User A enables Flight mode and backgrounds the app
3. User B sends seven messages in a row ```T1, T2, T3, T4, T5, T6, T7```
4. User A disables Flight mode, waits until network is established and re-opens the App
5. See if all 7 messages are delivered to User A

**Actual result:** 
User A received only 2 out of 7 messages sent by User B - ```T1, T7```
The rest 5 messages ```T2, T3, T4, T5, T6``` remained not delivered to User A. I have been waiting for half an hour, performed multiple re-logins by both users, sent some more messages in channel but ```T2, T3, T4, T5, T6``` remained not delivered. 

[User_A_Android_receiver.zip](https://github.com/status-im/status-mobile/files/13416971/User_A_Android_receiver.zip)
[User_B_IOS_sender.zip](https://github.com/status-im/status-mobile/files/13416972/User_B_IOS_sender.zip)

https://github.com/status-im/status-mobile/assets/97245802/cac4fbc8-3d8f-4dab-969f-b8a4053c2e9b


![photo_2023-11-20 17 58 00](https://github.com/status-im/status-mobile/assets/97245802/eeeeacde-53b8-4d42-8ea1-fae235d93168)
![photo_2023-11-20 17 58 03](https://github.com/status-im/status-mobile/assets/97245802/6fe32dc3-75ea-4611-9e2f-de14dfc3f22a)

**Expected result:** messages are delivered to the receiver

**Comment by pavloburykh** [comment]
closing this one as opened a new ticket https://github.com/status-im/status-mobile/issues/18031
